src/audiences/ui/components/rule.js

```diff
@@ -64,14 +64,24 @@ const RuleInput = props => {
 	} = props;
 
 	// Get the currently available values for the field.
-	const data = ( currentField.data && currentField.data.map( datum => datum.value ) ) || [];
+	const options = ( currentField.data && currentField.data.map( option => ( {
+		value: option.value,
+		label: ( option?.label ? `${ option.label } (${ option.value })` : option.value ) + String( option?.percent ? ` ~${ option.percent }%` : '' ),
+		count: Number( option?.count ) || 0,
+		percent: Number( option?.percent ) || 0,
+	} ) ) ) || [];
+	// Sort options by their size/hits.
+	const topHits = options.filter( option => option.percent ).slice( 0, 10 );
+	topHits.sort( ( a, b ) => b.percent - a.percent );
+	// Check if the filter doesn't allow free text.
+	const allowFreeText = ! currentField?.options?.disable_free_text;
 
 	// Get input refs to manage focus.
 	const inputEl = useRef( null );
 	const dropdownRef = useRef( null );
 
 	// Default to display of the dropdown if the value is in existing data or empty.
-	const defaultDropdownState = value === '' || data.indexOf( value ) >= 0;
+	const defaultDropdownState = value === '' || options.find( option => option.value === value );
```

Comment:
Hm, what's the reason for using `find` instead? This is intended to be a boolean value but `find` will return the matching item from the array I thought.

Reply:
True, needs casting here. It still worked as the casting happens below, and null is falsey.
```suggestion
	const defaultDropdownState = value === '' || !! options.find( option => option.value === value );
```